File: tools/_shared/inc/models/gtfs_static/stop.py
import sqlite3

class Stop:
    def __init__(self, stop_id: str, stop_name: str, stop_lon, stop_lat, location_type, parent_station):
        self.stop_id = stop_id
        self.stop_name = stop_name
        self.stop_lon = stop_lon
        self.stop_lat = stop_lat
        self.location_type = location_type
        self.parent_station = parent_station

    @staticmethod
    def init_from_db_row(db_row: sqlite3.Row):
        stop_id = db_row['stop_id']
        stop_name = db_row['stop_name']
        stop_lon = db_row['stop_lon']
        stop_lat = db_row['stop_lat']

        # patch the raw sqlite3.Row objects to allow dict lookups
        if 'get' not in dir(db_row):
            db_row = dict(db_row)

        location_type = db_row.get('location_type') or None
        parent_station = db_row.get('parent_station') or None

        entry = Stop(stop_id, stop_name, stop_lon, stop_lat, location_type, parent_station)

        return entry
        

File: apps/hrdf-check-duplicates/js/ts/models/gtfs/stop.ts
import { HRDF_Stop_DB } from "../../types/hrdf/stop_db"

export default class Stop {
    public stop_id: string
    public stop_name: string
    public stop_lon: number
    public stop_lat: number
    public location_type: string | null
    public parent_station: string | null

    constructor(stop_id: string, stop_name: string, stop_lon: number, stop_lat: number) {
        this.stop_id = stop_id
        this.stop_name = stop_name
        this.stop_lon = stop_lon
        this.stop_lat = stop_lat

        this.location_type = null
        this.parent_station = null
    }

    public static initFromHRDFStopDB(stopDB: HRDF_Stop_DB) {
        const stop_id = stopDB.stop_id
        const stop_name = stopDB.stop_name
        const stop_lon = stopDB.stop_lon
        const stop_lat = stopDB.stop_lat

        const stop = new Stop(stop_id, stop_name, stop_lon, stop_lat)
        return stop
    }
}